src/providers/beeminder-api.ts

```diff
@@ -60,6 +61,13 @@ export class BeeminderApi {
              .catch(error => Observable.throw(error.json().error));
   }
 
```

Comment:
This should be a post request. So you can have a request body instead of having all those nasty parameters in the url.